### DISCUSSION TITLE
switching from portainer

### DISCUSSION THREAD

**Discussion by MrBabafemi** [louislam/dockge#624] [Ask for Help]
### ⚠️ Please verify that this bug has NOT been raised before.

- [X] I checked and didn't find similar issue

### 🛡️ Security Policy

- [X] I agree to have read this project [Security Policy](https://github.com/louislam/dockge/security/policy)

### 📝 Describe your problem

hello all.  recently been told about this great app and i want to use it but i need to verify some things first.  so in order to switch from portainer to dockge i need to stop the container in portainer, then copy the yml over to dockge? do i need to remove the containers altogether from docker before i start them up in dockge?  when i tried to start the stacks in dockge it gave me errors about already existing containers and removing or renamimg them before trying again and the instructions i had read hadnt mentioned the removing/renaming of the existing containers.  im assuming this would be the correct method but i wanted to verify first.

thanks in advance for any assistance given,

LAG

### 🐻 Dockge Version

Version: 1.4.2

### 💻 Operating System and Arch

Ubuntu 20.04.6 LTS

### 🌐 Browser

Google Chrome

### 🐋 Docker Version

latest...

**Comment by MrBabafemi** [comment]
ok seems i had the right idea and went and tried it the way i described above.  it worked for the most part.  only two containers didnt come right back up as they were. qbittorrent and bazaar.  qbittorrent isnt excepting the usn/pwd and bazarr is looking to be newly configured. any ideas why those two didnt just restart with the original setings still in place?

**Comment by Celer21** [comment]
Did you copy .env for those two containers? 
Are you sure paths in volumes are correct and same as in portainer?

**Reply by MrBabafemi** [reply]
I stopped and removed the containers in portainer. Copied the docker.yml file into dockge, updated and hit start.  Never saw any directions saying to copy/move any files. For the most part it worked perfectly except for a few containers.  For bazarr I had to input the configuration from scratch. For qbtorrent I had to get the admin pwd from the log and change username and pwd.  My streaming stack I left in portainer. That has radarr, sonarr, readarr and jellyfin.  It's not a perfect setup but everything is working  properly for now and I'll move that last stack when I figure out why it didn't work this time.

**Reply by shalafi99** [reply]
> I stopped and removed the containers in portainer. Copied the docker.yml file into dockge, updated and hit start. Never saw any directions saying to copy/move any files. For the most part it worked perfectly except for a few containers. For bazarr I had to input the configuration from scratch. For qbtorrent I had to get the admin pwd from the log and change username and pwd. My streaming stack I left in portainer. That has radarr, sonarr, readarr and jellyfin. It's not a perfect setup but everything is working properly for now and I'll move that last stack when I figure out why it didn't work this time.

Regarding your original question, I think it would be advisable to delete any containers created by a stack in portainer before firing up that same stack from dockge to avoid getting a "This stack is not managed by Dockge." message.
Since bind mounts and defined volumes would persist, there should be no issue there. I did such a move not too long ago.

But since you had to redo one configuration from scratch (bazarr), the only reason which occurs to me would be the "new" stack (inside dockge) was not able to leverage your original volume or bind mount defined in the "old" stack (in portainer).
Considering "new" and "old" in this context above refers to the exact same compose file it should not happen🤷🏻‍♂️ 

Are you using a bind mount or volume with bazaar for its /config?

**Reply by MrBabafemi** [reply]
I believe I am using a volume for my media files (had to look up the difference between volumes and bind mounts).  All my yml files are pointed towards the media  files saved on my raid5 nas.   And I had stopped and removed the containers after I had the initial issue once I realized what the problem was. Which later perplexed me once I had issues when attempting to start the containers in dockge.  If you need any info to assist me hopefully I can get it for you.

**Reply by shalafi99** [reply]
I am not that much knowledgeable myself, but typically a bind mount parameter begins with either a "./" or a "/" which references respectively either a relative or absolute path your docker host already has access to.

While a volume begins with neither of those and can have different parameters depending on the kind of mount it is, such as mounting the shares/volumes from your NAS like you seem to be doing for media files.

But what about the place where the config data is stored for each of those 4 containers defined in your media stack (**edit**: _for all but jellyfin, I understand it would be the one volume which gets mapped as /config inside the container_) , those are also volumes defined to use something offered by the NAS?

**Reply by MrBabafemi** [reply]
Yes after I sent previous msg I found that I actually do have bind mounts.  Sorry I sometimes have trouble using the proper names or terminology.  All yml files are saved in their proper stack folders ie streaming or downloading.  No settings have changed or been moved after I stopped/removed the containers in portainer. 

Before I stopped/removed them I had copied the yml into dockge but did not hit start in dockge until after I verified that the container was removed in terminal using docker  ps -a. Then I hit update and start in dockge a minute later.

**Reply by shalafi99** [reply]
That said, did you figure out why you had to reconfigure bazaar? And thus would know what to avoid while migrating the stack with the other "arrs".
An idea about a "pre-test" of sorts for the mounts your containers will need came to mind, but after your explanation I began to suspect you have it under control now. Please let me know.

**Reply by MrBabafemi** [reply]
I do not know why I had to reconfigured it.  But I didn't want to take a chance with the rest of the -arrs. Losing subtitles is one thing loosing media is another.  Which is why I wanted to make sure I hadn't missed a step.  All the other containers swapped over with no issues at all.  But I did them in the order of less important to most important. 

Moved first... nocodb and mysql
Moved 2nd... heimdall motioneye prowlar metube  home assistant readarr
Bith of these went smoothly no issues at all.

Moved 3rd... qbittorent bazarr jackett
Qb gave me usn/pwd issues... got back in eventually
Bazarr had to be reconfigured... back to normal now
Jackett... I had stopped using when I switched to prowlarr but I added it back in just in case. (It kept disconnecting under portainer figured i'd see what it does in dockge)

Move 4... sonarr radarr jellyfin. They came up similar to bazarr as if they needed to be reconfigured.  So I put them back into portainer and will leave them there until I figure out the problem. 

I did each stack exactly the same.  I'd be willing to try it again if the logs are needed to assist me.

**Reply by shalafi99** [reply]
When you brought the stack up in dockge, did you check inside the container's consoles if the data contained in each bind mount was accessible?

Something which came to mind - perhaps dockge is not being able to interpret a compose element due to syntax. Causing this difference between portainer and dockge.

Not specifically to your case here, but I have seen other discussions/issues related to compose syntax/formatting. An example (which is not functionally problematic and/or breaks usage, though) is #176 

Since your issue seems to be related to how the containers created by dockge access their mounted data (_I cannot think of another reason for radarr requiring a reconfiguration from scratch for example, other than it not being able to access the existing config in the first place_), I suppose that only further controlled testing would be able to expose what exactly.

I thought of asking for the "volumes" section from your compose, but my containers use only docker volumes (no NAS in my setup) I would not be able to easily attempt to reproduce the issue, if at all (thought of that because I have both portainer CE and dockge here, despite only using portainer at the moment)
